### PR TITLE
Error SPI implementation

### PR DESCRIPTION
**Pull request by MXPOL**
This PR refactors the errors class and introduces a new `CollectionChangeNotSupported` error for MySQL, PostgreSQL, and MSSQL.

A few important notes about the new error:

- NoSQL databases will never throw the `CollectionChangeNotSupported` error.
- While Spanner can support partial column type changes, this functionality is not supported in the adapter due to the lack of support for many of the column changes that the adapter offers in Spanner.

### CODE REVIEW

File: apps/velo-external-db/test/e2e/app_data_hooks.e2e.spec.ts [hunk 1]
```diff
@@ -518,14 +518,13 @@ describe(`Velo External DB Data Hooks: ${currentDbImplementationName()}`, () =>
                 dataHooks: {
                     beforeAll: (_payload, _requestContext, _serviceContext) => {
                         const error = new Error('message')
-                        error['status'] = '409'
```
Comment (MXPOL) [inline apps/velo-external-db/test/e2e/app_data_hooks.e2e.spec.ts:521]: I don't think it's right to set status to error ... I think an internal error (status 500) should be thrown every time the client code in hooks throws.

File: apps/velo-external-db/test/e2e/app_data_hooks.e2e.spec.ts [hunk 2]
```diff
@@ -518,14 +518,13 @@ describe(`Velo External DB Data Hooks: ${currentDbImplementationName()}`, () =>
...
                         throw error
                     }
                 }
             })
 
             await expect(axios.post('/items/remove', hooks.writeRequestBodyWith(ctx.collectionName, [ctx.item]), authOwner)).rejects.toMatchObject(
-                errorResponseWith(409, 'message')
+                errorResponseWith('WDE0054', 'message')
```
Comment (MXPOL) [inline apps/velo-external-db/test/e2e/app_data_hooks.e2e.spec.ts:527]: WDE0054 stands for unknown error

File: libs/velo-external-db-core/src/web/domain-to-spi-error-translator.ts [hunk 3]
```diff
@@ -1,48 +1,56 @@
...
-        return ErrorMessage.invalidProperty(fieldDoesNotExist.collectionName, fieldDoesNotExist.propertyName, fieldDoesNotExist.message)
+        const fieldDoesNotExist = err as domainErrors.FieldDoesNotExist
+        return new InvalidPropertyError(fieldDoesNotExist.collectionName, fieldDoesNotExist.propertyName, fieldDoesNotExist.message)
   
-      case domainErrors.UnsupportedSchemaOperation:
-        const unsupportedSchemaOperation: domainErrors.UnsupportedSchemaOperation = err
-        return ErrorMessage.operationIsNotSupportedByCollection(unsupportedSchemaOperation.collectionName, unsupportedSchemaOperation.operation, unsupportedSchemaOperation.message)
+      // case domainErrors.UnsupportedSchemaOperation:
+      //   const unsupportedSchemaOperation = err as domainErrors.UnsupportedSchemaOperation
+      //   return ErrorMessage.operationIsNotSupportedByCollection(unsupportedSchemaOperation.collectionName, unsupportedSchemaOperation.operation, unsupportedSchemaOperation.message)
```
Comment (MXPOL) [inline libs/velo-external-db-core/src/web/domain-to-spi-error-translator.ts:24]: There is no Wix error for such an error, and I'm not sure it's necessary.
What do you think?

File: apps/velo-external-db/test/e2e/app_data_hooks.e2e.spec.ts (hunk 1 above)
Comment (Idokah) [reply apps/velo-external-db/test/e2e/app_data_hooks.e2e.spec.ts:521]: why?

File: apps/velo-external-db/test/e2e/app_schema.e2e.spec.ts [hunk 4]
```diff
@@ -112,6 +113,21 @@ describe(`Schema REST API: ${currentDbImplementationName()}`,  () => {
             await axiosClient.post('/collections/delete', { collectionId: ctx.collectionName }, authOwner)
             await expect(schema.retrieveSchemaFor(ctx.collectionName, authOwner)).rejects.toThrow('404')
         })
+
+        describe('error handling', () => {
+            if (['mysql', 'postgres', 'mssql'].includes(currentDbImplementationName())) {
```
Comment (Idokah) [inline apps/velo-external-db/test/e2e/app_schema.e2e.spec.ts:118]: lets at least name this array as a constant

File: apps/velo-external-db/test/e2e/app_schema.e2e.spec.ts [hunk 5]
```diff
@@ -112,6 +113,21 @@ describe(`Schema REST API: ${currentDbImplementationName()}`,  () => {
...
+                test('should throw "CollectionChangeNotSupported" error on failed collection alteration', async() => {
```
Comment (Idokah) [inline apps/velo-external-db/test/e2e/app_schema.e2e.spec.ts:119]: testIfSupportedOperationsIncludes(supportedOperations, [ ChangeColumnType ])

File: apps/velo-external-db/test/e2e/app_schema_hooks.e2e.spec.ts [hunk 6]
```diff
@@ -251,14 +251,13 @@ describe(`Velo External DB Schema Hooks: ${currentDbImplementationName()}`, () =
                 schemaHooks: {
                     beforeAll: (_payload, _requestContext, _serviceContext) => {
                         const error = new Error('message')
-                        error['status'] = '409'
```
Comment (Idokah) [inline apps/velo-external-db/test/e2e/app_schema_hooks.e2e.spec.ts:254]: here as well, lets give the ability to the user to set the error status

File: libs/velo-external-db-core/src/web/domain-to-spi-error-translator.ts (hunk 3 above)
Comment (Idokah) [reply libs/velo-external-db-core/src/web/domain-to-spi-error-translator.ts:24]: we are still throwing it in some places (schema.ts) - so or throw other error, or catch it and translate it to some error which is not internal.

**Review by Idokah — APPROVED** [review]

**Pull request merged**